src/picongpu/include/fields/background/templates/bgrTWTS.hpp

```diff
@@ -0,0 +1,659 @@
+                 *  Bz --> -Bx
+                 */
+                eFieldPositions_SI[i] = float3_64( -(eFieldPositions_SI[i]).z(), (eFieldPositions_SI[i]).y(), (eFieldPositions_SI[i]).x() );
+                
+                /*  Since, the laser propagation direction encloses an angle of phi with the simulation y-axis (i.e. direction of sliding window),
+                 *  the positions vectors are rotated around the simulation x-axis before calling the TWTS field functions. Note: The TWTS field
+                 *  functions are in non-rotated frame and only use the angle phi to determine the required amount of pulse front tilt.
+                 *  RotationMatrix[PI/2+phi].(y,z) (180Deg-flip at phi=90Deg since coordinate system in paper is oriented the other way round.) */
+                
+                /* Note: The x-axis of rotation is fine in 2D, because that component now contains the (non-existing) simulation z-coordinate. */
+                eFieldPositions_SI[i] = float3_64( (eFieldPositions_SI[i]).x(),
+                                                   -sin(phi)*(eFieldPositions_SI[i]).y()-cos(phi)*(eFieldPositions_SI[i]).z(),
+                                                   +cos(phi)*(eFieldPositions_SI[i]).y()-sin(phi)*(eFieldPositions_SI[i]).z()  );
+            }
+            
+            return eFieldPositions_SI;
+        }
+        
+        HDINLINE float_64
+        TWTSFieldE::getTime_SI(const uint32_t currentStep) const
+        {
+            const float_64 time_SI = precisionCast<float_64>(currentStep) * dt;
+                       
+            if ( auto_tdelay ) {
+                
+                // angle between the laser pulse front and the y-axis. (Good approximation for beta0 \simeq 1. For exact relation look in TWTS core routines for Ex, By or Bz.)
+                const float_64 eta = PI/2 - (phi/2);
+                /* halfSimSize[2] --> Half-depth of simulation volume (in z); By geometric projection we calculate the y-distance walkoff
+                 * of the TWTS-pulse. The abs()-function is for correct offset for -phi<-90Deg and +phi>+90Deg. */
+                const float_64 y1=precisionCast<float_64>(halfSimSize[2]*picongpu::SI::CELL_DEPTH_SI)*abs(cos(eta)); 
+                const float_64 m=3.; // Fudge parameter to make sure, that TWTS pulse starts to impact simulation volume at low intensity values.
+                const float_64 y2=m*(pulselength_SI*picongpu::SI::SPEED_OF_LIGHT_SI)/cos(eta); // Approximate cross section of laser pulse through y-axis, scaled with "fudge factor" m.
+                const float_64 y3=focus_y_SI; // y-position of laser coordinate system origin within simulation.
+                const float_64 tdelay= (y1+y2+y3)/(picongpu::SI::SPEED_OF_LIGHT_SI*beta_0);
+                
+                return time_SI-tdelay;
+            }
+            else
+                return time_SI-tdelay_user_SI;
+        }
+        
+        template<>
+        HDINLINE float3_X
```

Comment:
Since the result is a normalized quantity the datatype is correct, but I renamed the function.